go/test/endtoend/vtgate/seq_test.go

```diff
@@ -0,0 +1,174 @@
+
+	"vitess.io/vitess/go/mysql"
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/test/endtoend/cluster"
+)
+
+var (
+	ClusterInstance *cluster.LocalProcessCluster
+	KeyspaceName    = "ks"
+	Cell            = "zone1"
+	Hostname        = "localhost"
+	SQLSchema       = `
+	create table sequence_test(
+		id bigint,
+		val varchar(16),
+		primary key(id)
+	)Engine=InnoDB;
+
+	create table sequence_test_seq (
+		id int default 0, 
+		next_id bigint default null, 
+		cache bigint default null, 
+		primary key(id)
+	) comment 'vitess_sequence' Engine=InnoDB;
+	`
+
+	VSchema = `
+		{	
+			"sharded":false,
+			"vindexes": {
+				"hash_index": {
+					"type": "hash"
+				}
+			},	
+			"tables": {
+				"sequence_test":{
+					"auto_increment":{
+						"column" : "id",
+						"sequence" : "sequence_test_seq"
+					},
+					"column_vindexes": [
+						{
+							"column": "id",
+							"name": "hash_index"
+						}
+					]
+				},
+				"sequence_test_seq": {
+					"type":   "sequence"
+				}
+			}
+		}
+	`
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+
+	exitCode := func() int {
+		ClusterInstance = &cluster.LocalProcessCluster{Cell: Cell, Hostname: Hostname}
+		defer ClusterInstance.Teardown()
+
+		// Start topo server
+		if err := ClusterInstance.StartTopo(); err != nil {
+			return 1
+		}
+
+		// Start keyspace
+		keyspace := &cluster.Keyspace{
+			Name:      KeyspaceName,
+			SchemaSQL: SQLSchema,
+			VSchema:   VSchema,
+		}
+		if err := ClusterInstance.StartUnshardedKeyspace(*keyspace, 1, false); err != nil {
+			return 1
+		}
+
+		// Start vtgate
+		if err := ClusterInstance.StartVtgate(); err != nil {
+			return 1
+		}
+
+		return m.Run()
+	}()
+	os.Exit(exitCode)
+}
+
+func exec(t *testing.T, conn *mysql.Conn, query string) *sqltypes.Result {
+	t.Helper()
+	qr, err := conn.ExecuteFetch(query, 1000, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return qr
+}
+
+func TestSeq(t *testing.T) {
+	ctx := context.Background()
+	vtParams := mysql.ConnParams{
+		Host: "localhost",
+		Port: ClusterInstance.VtgateMySQLPort,
+	}
+	conn, err := mysql.Connect(ctx, &vtParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	//Initialize seq table
+	exec(t, conn, "insert into sequence_test_seq(id, next_id, cache) values(0,1,10)")
+
+	//Insert 4 values in the main table
+	exec(t, conn, "insert into sequence_test(val) values('a'), ('b') ,('c'), ('d')")
+
+	// Test select calls to main table and verify expected id.
+	qr := exec(t, conn, "select id, val  from sequence_test where id=4")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[[INT64(4) VARCHAR("d")]]`; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+
+	// Test next available seq id from cache
+	qr = exec(t, conn, "select next 1 values from sequence_test_seq")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[[INT64(5)]]`; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+
+	//Test next_id from seq table. This will be alloted in case cache is blew up.
```

Comment:
Can you fix this comment to say exactly what next_id is supposed to represent/ return?
Something like `As soon as we start using values from the cache, next_id is updated to the last value in the cache + 1` (or whatever is correct).

Reply:
Done